### PR TITLE
Migrate from gtest to doctest

### PR DESCRIPTION
**Pull request by elliottt**
### Motivation

gtest makes it easy to do the wrong thing, see #2968.

### Test plan

See included automated tests.

### CODE REVIEW

**Comment by DarkDimius** [comment]
Could we finish the migration before merging to not be in the in-between state?

**Comment by jvilk-stripe** [comment]
Sure, we can do that.

File: namer/test/namer_test.cc [hunk 1]
```diff
@@ -1,4 +1,5 @@
-#include "gtest/gtest.h"
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
```
[the remark this answers is not in the document]
Comment (elliottt) [reply namer/test/namer_test.cc]: I moved this into the `@doctest//:doctest_main` library, and marked that library as `testonly`.

File: test/cli_test_runner.cc [hunk 2]
```diff
@@ -0,0 +1,518 @@
+#include "doctest.h"
```
Comment (DarkDimius) [inline test/cli_test_runner.cc:1]: The name of this file is confusing. It doesn't run `cli` tests.
test_corpus_runner?

Comment (elliottt) [reply test/cli_test_runner.cc:1]: Sure, happy to change it.

Comment (elliottt) [reply test/cli_test_runner.cc:1]: Actually, how about `pipeline_test_runner.cc`? This file doesn't run the LSP or Parser tests anymore, so I'd like it to have a more specific name.

Comment (DarkDimius) [reply test/cli_test_runner.cc]: Sure!

**Pull request merged**